pandas/_libs/src/parser/tokenizer.c

```diff
@@ -1834,201 +1839,196 @@ int uint64_conflict(uint_state *self) {
-    ++p;
+  if (!p_item || *p_item == '\0') {
+    *error = ERROR_NO_DIGITS;
+    return 0;
   }
 
-  // Handle sign.
-  const bool isneg = *p == '-' ? true : false;
-  // Handle sign.
-  if (isneg || (*p == '+')) {
-    p++;
+  while (isspace_ascii(*p_item)) {
+    ++p_item;
   }
 
-  // Check that there is a first digit.
-  if (!isdigit_ascii(*p)) {
-    // Error...
+  if (!has_digit_int(p_item)) {
     *error = ERROR_NO_DIGITS;
     return 0;
   }
 
-  int64_t number = 0;
-  if (isneg) {
-    // If number is greater than pre_min, at least one more digit
-    // can be processed without overflowing.
-    int dig_pre_min = -(int_min % 10);
-    int64_t pre_min = int_min / 10;
-
-    // Process the digits.
-    char d = *p;
-    if (tsep != '\0') {
-      while (1) {
-        if (d == tsep) {
-          d = *++p;
-          continue;
-        } else if (!isdigit_ascii(d)) {
-          break;
-        }
-        if ((number > pre_min) ||
-            ((number == pre_min) && (d - '0' <= dig_pre_min))) {
-          number = number * 10 - (d - '0');
-          d = *++p;
-        } else {
-          *error = ERROR_OVERFLOW;
-          return 0;
-        }
-      }
-    } else {
-      while (isdigit_ascii(d)) {
-        if ((number > pre_min) ||
-            ((number == pre_min) && (d - '0' <= dig_pre_min))) {
-          number = number * 10 - (d - '0');
-          d = *++p;
-        } else {
-          *error = ERROR_OVERFLOW;
-          return 0;
-        }
-      }
-    }
-  } else {
-    // If number is less than pre_max, at least one more digit
-    // can be processed without overflowing.
-    int64_t pre_max = int_max / 10;
-    int dig_pre_max = int_max % 10;
-
-    // Process the digits.
-    char d = *p;
-    if (tsep != '\0') {
-      while (1) {
-        if (d == tsep) {
-          d = *++p;
-          continue;
-        } else if (!isdigit_ascii(d)) {
-          break;
-        }
-        if ((number < pre_max) ||
-            ((number == pre_max) && (d - '0' <= dig_pre_max))) {
-          number = number * 10 + (d - '0');
-          d = *++p;
-
-        } else {
-          *error = ERROR_OVERFLOW;
-          return 0;
-        }
-      }
-    } else {
-      while (isdigit_ascii(d)) {
-        if ((number < pre_max) ||
-            ((number == pre_max) && (d - '0' <= dig_pre_max))) {
-          number = number * 10 + (d - '0');
-          d = *++p;
+  char buffer[PROCESSED_WORD_CAPACITY];
+  size_t str_len = strlen(p_item);
+  if (tsep != '\0' && memchr(p_item, tsep, str_len) != NULL) {
+    int status = copy_string_without_char(buffer, p_item, str_len, tsep);
 
-        } else {
-          *error = ERROR_OVERFLOW;
-          return 0;
-        }
-      }
+    if (status != 0) {
+      // Word is too big, probably will cause an overflow
+      *error = ERROR_OVERFLOW;
+      return 0;
     }
+
+    p_item = buffer;
   }
 
-  // Skip trailing spaces.
-  while (isspace_ascii(*p)) {
-    ++p;
+  if (errno == ERANGE) {
```

Comment:
This is lost in a refactor right? I don't think errno will be set by anything preceding this?